Automattic/wp-calypso
# Plans: Remove get-plan-prices selector

Part of addressing https://github.com/Automattic/wp-calypso/issues/86638

## Proposed Changes

Follow up on https://github.com/Automattic/wp-calypso/pull/89396 to remove the `getPlanPrices` selector from the codebase altogether.

Some tests could be migrated to the `data-stores` selector, although they don't look all that valuable. 

## Testing Instructions



- tests/build passing
- there should be no longer references to this selector, so it should be safe to remove

## Pre-merge Checklist
